docs/content/en/integration/openid-connect/roundcube/index.md

```diff
@@ -0,0 +1,231 @@
+$config['oauth_token_uri'] = 'https://auth.example.com/api/oidc/token';
+$config['oauth_identity_uri'] = 'https://auth.example.com/api/oidc/userinfo';
+$config['oauth_identity_fields'] = ['email'];
+$config['oauth_scope'] = 'email openid profile';
+// Optionally, skip Roundcube's login page
+//$config['oauth_login_redirect'] = true;
+
+// Note on IMAP and SMTP backends (cf. Debian's '/etc/roundcube/defaults.inc.php').
+// - For an IMAP instance on localhost, the default conf is enough. Otherwise,
+//   set the corresponding SSL/TLS options via 'imap_host' and 'imap_conn_options';
+// - For an SMTP instance on localhost, no auth would be required. However
+//   RC's OAuth enforces 'smtp_auth_type' = 'XOAUTH2' plus credentials (cf.
+//   <https://github.com/roundcube/roundcubemail/issues/9183>), thus you *must*
+//   use TLS or SSL via 'smtp_host' and 'smtp_conn_options'!
+```
+
+### Authelia
+
+The following YAML configuration is an example __Authelia__
+[client configuration](../../../configuration/identity-providers/openid-connect/clients.md) for use with [Roundcube]
+which will operate with the above example:
+
+```yaml
+identity_providers:
+  oidc:
+    ## The other portions of the mandatory OpenID Connect 1.0 configuration go here.
+    ## See: https://www.authelia.com/c/oidc
+    clients:
+    - id: 'roundcube'
+      description: 'Roundcube'
+      secret: '$pbkdf2-sha512$310000$c8p78n7pUMln0jzvd4aK4Q$JNRBzwAo0ek5qKn50cFzzvE9RXV88h1wJn5KGiHrD0YKtZaR/nCb2CJPOsKaPK0hjf.9yHxzQGZziziccp6Yng'  # The digest of 'insecure_secret'.
+      public: false
+      authorization_policy: 'two_factor'
+      redirect_uris:
+        - 'https://roundcube.example.com/oauth/callback/'
+      scopes:
+        - 'openid'
+        - 'profile'
+        - 'email'
+# optional
+access_control:
+  rules:
+    - domain: 'webmail.example.com'
+      resources:
+        - '^/(skins|plugins|jquery|program)([/?].*)?$'
+      policy: bypass
+```
+
+Also, mind that your reverse proxy might need special arrangements -- see an example for Nginx here below.
+
+### Nginx
+
+The [standard authrequest-based
+example](https://www.authelia.com/integration/proxies/nginx/#standard-example)
+applies. However, if you want to use the *same* reverse-proxy instance for
+your protected application, its corresponding vhost might need the following
+adaptations:
+
+```nginx
+# Backend proxied app on the same host
+server {
+    # See N.B. below
+    listen 9092;
+    server_name roundcube;
+
+    ...
+
+    location ~ [^/]\.php(?:/|$) {
+        fastcgi_split_path_info ^(.+?\.php)(/.*)$;
+        if (!-f $document_root$fastcgi_script_name) {
+            return 404;
+        }
+
+        include fastcgi_params;
+
+        fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
+        fastcgi_param PATH_INFO       $fastcgi_path_info;
+
+        # See N.B. below
+        fastcgi_param  SERVER_PORT '';
+
+        fastcgi_pass $your_php_upstream;
+    }
+}
+```
```

Comment:
I do not really like adding specific proxy configurations here unless they're specific to OIDC which I do not think this is. I'd rather explain the specific OIDC requirements and adapt them to be generic for all proxies.

Also we do not support mixing both auth-request and OIDC specifically unless there is a reason. Could you explain the reasoning why you used it?

Reply:
Oops, sorry, the opening note is indeed a kind of mix-up remnant from my initial tests with auth-request. However, with Nginx, Roundcube's vhost still needs proper path splitting, because the more (?) classic formulation without `fastcgi_split_path_info` causes a redirection loop. I'll correct and resubmit.

Reply:
No worries. So is the old nginx config something unique to a particular guide or container which has since been updated? Like maybe in a situation where nginx is always used? Maybe it might be useful then.

Reply:
I reviewed my Nginx setup: this snippet can be dropped because the latest official Nginx wiki [covers most of it](https://www.nginx.com/resources/wiki/start/topics/examples/phpfcgi/), and the `SERVER_PORT` mod would be relevant only in the auth-req scenario.